File: src/game/powerups/BlockMoverPowerUp.java
package game.powerups;

import game.entity.blocks.MovableBlock;
import game.entity.models.BomberEntity;
import game.entity.models.Coordinates;
import game.utils.Paths;

import java.awt.image.BufferedImage;

public class BlockMoverPowerUp extends PowerUp {
    public BlockMoverPowerUp(Coordinates coordinates) {
        super(coordinates);
        getIncompatiblePowerUps().add(Hammer.class);
    }

    @Override
    public BufferedImage getImage() {
        return loadAndSetImage(Paths.getPowerUpsFolder() + "/hand.png");
    }

    @Override
    public int getDuration() {
        return 30;
    }

    @Override
    protected void doApply(BomberEntity entity) {
        entity.addClassInteractWithMouseDrag(MovableBlock.class);
    }

    @Override
    protected void cancel(BomberEntity entity) {
        entity.removeClassInteractWithDrag(MovableBlock.class);
    }
}

File: src/game/powerups/BlockMoverPowerUp.kt
package game.powerups

import game.entity.EntityTypes
import game.entity.blocks.MovableBlock
import game.entity.player.BomberEntity
import game.entity.models.Coordinates
import game.utils.Paths.powerUpsFolder
import java.awt.image.BufferedImage

class BlockMoverPowerUp : PowerUp {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    init {
        incompatiblePowerUps += HammerPowerUp::class.java
    }

    override fun getImage(): BufferedImage = loadAndSetImage("$powerUpsFolder/hand.png")

    override val duration: Int
        get() = 30

    override fun doApply(entity: BomberEntity) = entity.addClassInteractWithMouseDrag(MovableBlock::class.java)

    override fun cancel(entity: BomberEntity) = entity.removeClassInteractWithDrag(MovableBlock::class.java)

    override val type: EntityTypes
        get() = EntityTypes.BlockMoverPowerUp
}
